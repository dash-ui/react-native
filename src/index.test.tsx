import { render } from "@testing-library/react-native";
import * as React from "react";
import * as RN from "react-native";
import "@testing-library/jest-native/extend-expect";
import { createStyles } from "./index";

describe("styles()", () => {
  const { styles, styled } = createStyles({
    tokens: {
      spacing: [1, 2, 3, 4],
    } as const,
  });

  it("should pass", () => {
    const style = styles({
      default: `
        color: red;
      `,
      foo: {
        color: "blue",
      },
    });

    expect(style("default", "foo")).toMatchSnapshot();
  });

  it("should too pass", () => {
    const style = styles({
      default: ({ spacing }) => `
        margin: ${spacing[0]}px;
      `,
    });

    expect(style("default")).toMatchSnapshot();
  });

  it("should too pass 2", () => {
    const style = styles({
      default: ({ spacing }) => ({
        margin: spacing[0],
      }),
    });

    expect(style("default")).toMatchSnapshot();
  });

  it("should too pass 3", () => {
    const style = styles({
      default: ({ spacing }) => ({
        margin: spacing[0] + "px",
      }),
      other: {
        color: "blue",
      },
    });

    expect(style("default", "other")).toMatchSnapshot();
  });

  it("should also pass", () => {
    const StyledView = styled.View(({ spacing }) => ({
      margin: spacing[0],
    }));

    const view = render(
      <StyledView style="background-color: blue;" testID="foo" />
    );

    expect(view.getByTestId("foo")).toHaveStyle({
      backgroundColor: "blue",
      margin: 1,
    });
  });

  it("should also pass 3", () => {
    const StyledView = styled(RN.View, { backgroundColor: "blue" });
    const view = render(<StyledView testID="foo" />);

    expect(view.getByTestId("foo")).toHaveStyle({ backgroundColor: "blue" });
  });
});

describe("styles.one()", () => {
  const { styles } = createStyles({
    tokens: {
      spacing: [1, 2, 3, 4],
    },
  });

  it("should pass", () => {
    const style = styles.one`
      color: red;
    `;
    expect(style()).toMatchSnapshot();
  });

  it("should also pass", () => {
    const style = styles.one<RN.TextStyle>(() => ({
      color: "blue",
    }));

    expect(style()).toMatchSnapshot();
  });
});

describe("styles.cls()", () => {
  const { styles } = createStyles({
    tokens: {
      spacing: [1, 2, 3, 4],
    },
  });

  it("should pass", () => {
    const style = styles.cls`
      color: red;
    `;

    expect(style).toMatchSnapshot();
  });
});
