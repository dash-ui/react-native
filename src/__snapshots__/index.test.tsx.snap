// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`styles() should pass 1`] = `
Object {
  "color": "blue",
}
`;

exports[`styles() should too pass 1`] = `
Object {
  "marginBottom": 1,
  "marginLeft": 1,
  "marginRight": 1,
  "marginTop": 1,
}
`;

exports[`styles() should too pass 2 1`] = `
Object {
  "marginBottom": 1,
  "marginLeft": 1,
  "marginRight": 1,
  "marginTop": 1,
}
`;

exports[`styles() should too pass 3 1`] = `
Object {
  "color": "blue",
  "margin": "1px",
}
`;

exports[`styles.cls() should pass 1`] = `
Object {
  "color": "red",
}
`;

exports[`styles.one() should also pass 1`] = `
Object {
  "color": "blue",
}
`;

exports[`styles.one() should pass 1`] = `
Object {
  "color": "red",
}
`;
